daemon/lib/source/Dobby.cpp

```diff
@@ -1436,6 +1438,118 @@ void Dobby::wakeup(std::shared_ptr<AI_IPC::IAsyncReplySender> replySender)
     AI_LOG_FN_EXIT();
 }
 
+// -----------------------------------------------------------------------------
+/**
+ *  @brief mount a USB mass storage device inside container
+ *
+ *
+ *
+ *
+ */
+void Dobby::addMount(std::shared_ptr<AI_IPC::IAsyncReplySender> replySender)
+{
+    AI_LOG_FN_ENTRY();
+
+    // Expecting 4 arguments  (int32_t cd, string source, string target, int32_t mountFlags)
```

Comment:
I think the API should be `addMount(int32_t cd, string source, string target, vector<string> mountFlags, string mountData)`

Most notably the mount flags is a string list taking the same options as the mount command line app.  That's kinda what we've done for the spec file and potentially avoids different values for the mount flags.

Reply:
I was planning to use  syscall(SYS_mount_setattr,...) for the mount flags/attirbutes but it's not supported in kernel v5.4 Kirkstone devices have.

Reply:
using     MS_REMOUNT to set the mount flags instead of SYS_mount_setattr works.

Reply:
addMount() is now taking a string for the mountFlags argument. it's a comma seperated list of mount options (ro,bind etc).